File: java/ecbenchmark/src/ecbenchmark/Operator.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package ecbenchmark;

/**
 *
 * @author jcrada
 */
public abstract class Operator implements Function {

    protected Function innerFunction;

    public Operator() {
    }

    public Operator(Function innerFunction) {
        this.innerFunction = innerFunction;
    }

    public void setInnerFunction(Function f) {
        this.innerFunction = f;
    }

    public Function getInnerFunction() {
        return this.innerFunction;
    }
}

File: cpp/ecbenchmark/src/Operator.cpp
#include "ecbenchmark/Operator.h"

namespace ecb {

    Operator::Operator(Function* f) : _innerFunction(f) {

    }

    Operator::~Operator() {
        if (_innerFunction) delete _innerFunction;
    }

    void Operator::setInnerFunction(Function* f) {
        this->_innerFunction = f;
    }

    Function* Operator::innerFunction() const {
        return this->_innerFunction;
    }
    

}
